src/StoragelessSession/Session/Data.php

```diff
@@ -81,36 +136,30 @@ public static function newEmptySession(): self
         return new self([], []);
     }
 
-    /**
-     * @todo ensure serializable data?
-     */
-    public function set(string $key, $value)
+    public function has($key)
     {
-        $this->data[$key] = $value;
+        return isset($this->data[$key]);
     }
 
-    public function get(string $key)
+    public function isEmpty() : bool
     {
-        if (! $this->has($key)) {
-            throw new \OutOfBoundsException(sprintf('Non-existing key "%s" requested', $key));
+        foreach ($this->scope as $scope) {
+            if (! $scope->isEmpty()) {
+                return false;
+            }
         }
 
-        return $this->data[$key];
-    }
-
-    public function remove(string $key)
-    {
-        unset($this->data[$key]);
-    }
-
-    public function has(string $key): bool
-    {
-        return array_key_exists($key, $this->data);
+        return true;
     }
 
-    public function isEmpty()
+    public function isModified() : bool
```

Comment:
Why can't we just stop at the first modified scope? Do we really need to verify them all?